### PR TITLE
Bug#1583526 - added a note about double-escaping special chars

### PR DESCRIPTION
**Pull request by gaurav-nelson**
For https://bugzilla.redhat.com/show_bug.cgi?id=1583526

### CODE REVIEW

**Comment by gaurav-nelson** [comment]
@e-minguez PTAL

I have placed the following note:

> If you are using special characters (for example, `pass:[#]`, `{` or `}`) in the
> Ansible host inventory file, you need to double-escape them to make it work. For
> example, to use `pass:[mypasswordwith###hashsigns]` as a value for the variable
> `openshift_cloudprovider_openstack_password`, declare it as
> `openshift_cloudprovider_openstack_password='"mypasswordwith###hashsigns"'` in
> the Ansible host inventory file.

in following sections:

1. advanced_install > configuring-cluster-variables (PREVIEW: https://5b14a790b3127479ad7d3062--gnelson-preview.netlify.com/openshift-enterprise/(head%20detached%20at%20fetch_head)/install_config/install/advanced_install.html#configuring-cluster-variables)
2. configuring_openstack > openstack-configuring-masters-ansible (PREVIEW: https://5b14a790b3127479ad7d3062--gnelson-preview.netlify.com/openshift-enterprise/(head%20detached%20at%20fetch_head)/install_config/configuring_openstack.html#openstack-configuring-masters-ansible)

**Comment by gaurav-nelson** [comment]
@openshift/team-documentation PTAL

File: install_config/topics/escaping_special_characters.adoc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+Escaping special characters in ansible variables file
+
+This module included in the following assemblies:
+* install_config/configuring_openstack.adoc
+* install_config/install/advanced_install.adoc
+////
+
+[IMPORTANT]
+====
+If you are using special characters (for example, `pass:[#]`, `{` or `}`) in the
```
Comment (e-minguez) [inline install_config/topics/escaping_special_characters.adoc]: The syntax for an ini inventory file is ```pass=whatever``` instead ```pass:whatever```
I'd also use ```pass=mycomplexpa##word``` instead ```pass=[#]```

File: install_config/topics/escaping_special_characters.adoc [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Ansible host inventory file, you need to double-escape them to make it work. For
+example, to use `pass:[mypasswordwith###hashsigns]` as a value for the variable
```
Comment (e-minguez) [inline install_config/topics/escaping_special_characters.adoc]: Same: ```pass=mypasswordwith###hashsigns```

**Review by e-minguez — CHANGES_REQUESTED** [review]
I've added some comments. Thx!

File: install_config/topics/escaping_special_characters.adoc (hunk 1 above)
Comment (kalexand-rh) [inline install_config/topics/escaping_special_characters.adoc]: I might replace the first sentence: If a parameter value in the Ansible inventory file contains special characters, such as `#`, `{` or `}`, you must enclose the value in both single and double quotation marks.

File: install_config/topics/escaping_special_characters.adoc [hunk 3]
```diff
@@ -0,0 +1,17 @@
+////
```
Comment (kalexand-rh) [inline install_config/topics/escaping_special_characters.adoc]: @bfallonf, @vikram-redhat  what do you think of reusing a note like this?  Should we? Should we make a separate folder for notes?

**Review by kalexand-rh — COMMENTED** [review]
I have a question and a suggestion. This is good information!

File: install_config/topics/escaping_special_characters.adoc (hunk 1 above)
Comment (gaurav-nelson) [reply install_config/topics/escaping_special_characters.adoc]: My bad, the `pass:[...]` format is the asciidoc passthrough variable to render the text correctly.

The final rendered text will looks like this:
> If you are using special characters (for example, `#`, `{` or `}`) in the
> Ansible host inventory file, you need to double-escape them to make it work. For
> example, to use `mypasswordwith###hashsigns` as a value for the variable
> `openshift_cloudprovider_openstack_password`, declare it as
> `openshift_cloudprovider_openstack_password='"mypasswordwith###hashsigns"'` in
> the Ansible host inventory file.

**Comment by gaurav-nelson** [comment]
Thanks @kalexand-rh I have updated the text. However, I have kept the word `double-escaped` as it is a common word which devs understand.

**Comment by gaurav-nelson** [comment]
@e-minguez Can you please check the updates and confirm?
image

File: install_config/topics/escaping_special_characters.adoc (hunk 3 above)
Comment (vikram-redhat) [reply install_config/topics/escaping_special_characters.adoc]: @kalexand-rh isn't that just another module?

**Review by e-minguez — APPROVED** [review]
LGTM

**Comment by gaurav-nelson** [comment]
Changes already in 3.10 branch. Not cherry-picking.

**Pull request merged**